### PR TITLE
Smoothing gem

### PR DESCRIPTION
**Pull request by michalpelka**
Gem that allows to smooth movement of the entity.

### CODE REVIEW

File: Gems/Smoothing/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+o3de_gem_setup("Smoothing")
```
Comment (patrykantosz) [inline Gems/Smoothing/CMakeLists.txt:2]: I tested this PR on `point-release/23103` branch and I've had some problems with this line - I've had to copy something like this from some other gem (I believe that this is copied from RobotecSpectatorCamera gem):


```
o3de_find_ancestor_gem_root(gem_path gem_name "${CMAKE_CURRENT_SOURCE_DIR}")
if (NOT gem_name)
    set(gem_name "Smoothing")
endif()

if (NOT gem_path)
    set(gem_path ${CMAKE_CURRENT_SOURCE_DIR})
endif()

set(gem_json ${gem_path}/gem.json)

o3de_restricted_path(${gem_json} gem_restricted_path gem_parent_relative_path)

o3de_pal_dir(pal_dir ${CMAKE_CURRENT_SOURCE_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
```


I am not sure if above code is forward compatible, but it appears that `o3de_gem_setup` is not backward compatible or branch, that I've used to test this PR, is defective.

**Review by patrykantosz — APPROVED** [review]
Great job, smoothing component works as expected. Just some minor comments. Also I've noticed something strange - when I tried to enable this gem in my project I got something like this:
`[ERROR] o3de.enable_gem: Smoothing has the following dependency compatibility issues and requires the --force parameter to activate:
  The following dependency requirements could not be satisfied:
The project requires RobotecSpectatorCamera`
It's strange becasue `grep -rns ".*Spectator.*" ./Smoothing` returns empty list. Could you check this on your setup?

File: Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    explicit SmoothingComponentEditorComponent(const SmoothingConfig& configuration);
+
+    AZ_EDITOR_COMPONENT(SmoothingComponentEditorComponent, "{01910451-5821-799e-b17a-908d53847ff3}", SmoothingComponentEditorBase);
+    static void Reflect(AZ::ReflectContext* context);
+
+    // SmoothingComponentEditorBase interface overrides...
+    void Activate() override;
+    void Deactivate() override;
+    bool ShouldActivateController() const override;
+};
```
Comment (michalpelka) [inline Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.h:24]:
```suggestion
};

```

**Comment by michalpelka** [comment]
Thanks for review @patrykantosz ! I've updated PR, applied all your suggestions. I also squashed multiple commits.

File: Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        AZ_EDITOR_COMPONENT(SmoothingComponentEditorComponent, SmoothingComponentEditorComponentTypeId, SmoothingComponentEditorBase);
+        static void Reflect(AZ::ReflectContext* context);
+
+        // SmoothingComponentEditorBase interface overrides...
+        void Activate() override;
+        void Deactivate() override;
+        bool ShouldActivateController() const override;
+    };
+} // namespace Smoothing
```
Comment (patrykantosz) [inline Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.h:27]: New line.

File: Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    void SmoothingComponentEditorComponent::Deactivate()
+    {
+        SmoothingComponentEditorBase::Deactivate();
+    }
+
+    bool SmoothingComponentEditorComponent::ShouldActivateController() const
+    {
+        return true;
+    };
+} // namespace Smoothing
```
Comment (patrykantosz) [inline Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.cpp:46]: New line.

File: Gems/Smoothing/Code/Source/Clients/SmoothingComponent.h [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+        SmoothingComponent(const SmoothingConfig& config);
+        SmoothingComponent() = default;
+        ~SmoothingComponent() = default;
+
+        // Component overrides...
+        void Activate() override;
+        void Deactivate() override;
+    };
+} // namespace Smoothing
```
Comment (patrykantosz) [inline Gems/Smoothing/Code/Source/Clients/SmoothingComponent.h:92]: New line.

File: Gems/Smoothing/Code/Source/Clients/SmoothingComponent.cpp [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    void SmoothingComponent::Activate()
+    {
+        SmoothingComponentBase::Activate();
+    }
+
+    void SmoothingComponent::Deactivate()
+    {
+        SmoothingComponentBase::Deactivate();
+    }
+} // namespace Smoothing
```
Comment (patrykantosz) [inline Gems/Smoothing/Code/Source/Clients/SmoothingComponent.cpp:258]: New line.

**Review by patrykantosz — APPROVED** [review]

File: Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.h (hunk 3 above)
Comment (michalpelka) [reply Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.h:27]:
```suggestion
} // namespace Smoothing

```

File: Gems/Smoothing/Code/Source/Clients/SmoothingComponent.cpp (hunk 6 above)
Comment (michalpelka) [inline Gems/Smoothing/Code/Source/Clients/SmoothingComponent.cpp:258]:
```suggestion
} // namespace Smoothing

```

File: Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.cpp (hunk 4 above)
Comment (michalpelka) [inline Gems/Smoothing/Code/Source/Tools/SmoothingEditorComponent.cpp:46]:
```suggestion
} // namespace Smoothing

```

**Comment by michalpelka** [comment]
Are we going to merge it?

**Pull request merged**